src/hardware/serialport/softmodem.h

```diff
@@ -61,105 +61,102 @@ bool MODEM_ReadPhonebook(const std::string &filename);
 
 class CFifo {
 public:
-	CFifo(Bitu _size)
-		: data(_size),
-		  size(_size),
-		  pos(0),
-		  used(0)
-	{}
-
-	INLINE Bitu left(void) const {
-		return size - used;
-	}
-	INLINE Bitu inuse(void) const {
-		return used;
-	}
-	void clear(void) {
+	CFifo(uint32_t _size) : data(_size), size(_size), pos(0), used(0) {}
+
+	uint32_t left() const { return size - used; }
+	uint32_t inuse() const { return used; }
+	void clear()
+	{
 		used = 0;
 		pos = 0;
 	}
 
-	void addb(Bit8u _val) {
-		if(used >= size) {
-			static Bits lcount = 0;
+	void addb(uint8_t val)
+	{
+		if (used >= size) {
+			static uint16_t lcount = 0;
 			if (lcount < 1000) {
 				lcount++;
-				LOG_MSG("MODEM: FIFO Overflow! (addb)");
+				LOG_MSG("SERIAL: modem FIFO Overflow! (addb)");
 			}
 			return;
 		}
 		//assert(used<size);
-		Bitu where = pos + used;
+		size_t where = pos + used;
 		if (where >= size)
 			where -= size;
-		data[where]=_val;
-		//LOG_MSG("+%x",_val);
+		data[where] = val;
+		//LOG_MSG("+%x", val);
 		used++;
 	}
 
-	void adds(Bit8u * _str,Bitu _len) {
-		if ( (used+_len) > size) {
-			static Bits lcount=0;
+	void adds(uint8_t *str, size_t len)
+	{
+		if ((used + len) > size) {
+			static uint16_t lcount = 0;
 			if (lcount < 1000) {
 				lcount++;
-				LOG_MSG("MODEM: FIFO Overflow! (adds len %" PRIuPTR ")",
-				        _len);
+				LOG_MSG("SERIAL: modem FIFO Overflow! (adds len %u)",
+				        static_cast<uint32_t>(len));
```

Comment:
`static_cast<unsigned>(len)`

Reply:
Interesting; is `unsigned` equivalent to `uint32` (or is it loosely defined like `int`? I just want to be sure our type passed to `%u` will be handled correctly.

Reply:
It is just as loosely defined as `int` (and `u` is the correct formatting flag for unsigned). `uint32_t` values are formatted with `PRIu32`.

Reply:
Thanks for this; feels like `size_t` has essentially deprecated `unsigned int`?
`uint32_t` values are formatted with `PRIu32`: I'll take a pass through and fix these as well!

Reply:
`len` is now `size_t`' (matching the `pos` and `size` member variables, which `len` interacts with), and now printing `len` with `%zu`. Will see if MSYS2 handles it!